wolfBoot_stm32g0_v0.1/hal/stm32g0.c

```diff
@@ -0,0 +1,565 @@
+#define FLASH_SECTOR_4  0x0002000 /* 2 Kb   */
+#define FLASH_SECTOR_5  0x0002800 /* 2 Kb   */
+#define FLASH_SECTOR_6  0x0003000 /* 2 Kb   */
+#define FLASH_SECTOR_7  0x0003800 /* 2 Kb   */
+#define FLASH_SECTOR_8  0x0004000 /* 2 Kb   */
+#define FLASH_SECTOR_9  0x0004800 /* 2 Kb   */
+#define FLASH_SECTOR_10 0x0005000 /* 2 Kb   */
+#define FLASH_SECTOR_11  0x0005800 /* 2 Kb   */
+#define FLASH_SECTOR_12  0x0006000 /* 2 Kb   */
+#define FLASH_SECTOR_13  0x0006800 /* 2 Kb   */
+#define FLASH_SECTOR_14  0x0007000 /* 2 Kb   */
+#define FLASH_SECTOR_15  0x0007800 /* 2 Kb   */
+#define FLASH_SECTOR_16  0x0008000 /* 2 Kb   */
+#define FLASH_SECTOR_17  0x0008800 /* 2 Kb   */
+#define FLASH_SECTOR_18  0x0009000 /* 2 Kb   */
+#define FLASH_SECTOR_19  0x0009800 /* 2 Kb   */
+#define FLASH_SECTOR_20  0x000A000 /* 2 Kb   */
+#define FLASH_SECTOR_21  0x000A800 /* 2 Kb   */
+#define FLASH_SECTOR_22  0x000B000 /* 2 Kb   */
+#define FLASH_SECTOR_23  0x000B800 /* 2 Kb   */
+#define FLASH_SECTOR_24  0x000C000 /* 2 Kb   */
+#define FLASH_SECTOR_25  0x000C800 /* 2 Kb   */
+#define FLASH_SECTOR_26  0x000D000 /* 2 Kb   */
+#define FLASH_SECTOR_27  0x000D800 /* 2 Kb   */
+#define FLASH_SECTOR_28  0x000E000 /* 2 Kb   */
+#define FLASH_SECTOR_29  0x000E800 /* 2 Kb   */
+#define FLASH_SECTOR_30  0x000F000 /* 2 Kb   */
+#define FLASH_SECTOR_31  0x000F800 /* 2 Kb   */
+#define FLASH_SECTOR_32  0x0010000 /* 2 Kb   */
+#define FLASH_SECTOR_33  0x0010800 /* 2 Kb   */
+#define FLASH_SECTOR_34  0x0011000 /* 2 Kb   */
+#define FLASH_SECTOR_35  0x0011800 /* 2 Kb   */
+#define FLASH_SECTOR_36  0x0012000 /* 2 Kb   */
+#define FLASH_SECTOR_37  0x0012800 /* 2 Kb   */
+#define FLASH_SECTOR_38  0x0013000 /* 2 Kb   */
+#define FLASH_SECTOR_39  0x0013800 /* 2 Kb   */
+#define FLASH_SECTOR_40  0x0014000 /* 2 Kb   */
+#define FLASH_SECTOR_41  0x0014800 /* 2 Kb   */
+#define FLASH_SECTOR_42  0x0015000 /* 2 Kb   */
+#define FLASH_SECTOR_43  0x0015800 /* 2 Kb   */
+#define FLASH_SECTOR_44  0x0016000 /* 2 Kb   */
+#define FLASH_SECTOR_45  0x0016800 /* 2 Kb   */
+#define FLASH_SECTOR_46  0x0017000 /* 2 Kb   */
+#define FLASH_SECTOR_47  0x0017800 /* 2 Kb   */
+#define FLASH_SECTOR_48  0x0018000 /* 2 Kb   */
+#define FLASH_SECTOR_49  0x0018800 /* 2 Kb   */
+#define FLASH_SECTOR_50  0x0019000 /* 2 Kb   */
+#define FLASH_SECTOR_51  0x0019800 /* 2 Kb   */
+#define FLASH_SECTOR_52  0x001A000 /* 2 Kb   */
+#define FLASH_SECTOR_53  0x001A800 /* 2 Kb   */
+#define FLASH_SECTOR_54  0x001B000 /* 2 Kb   */
+#define FLASH_SECTOR_55  0x001B800 /* 2 Kb   */
+#define FLASH_SECTOR_56  0x001C000 /* 2 Kb   */
+#define FLASH_SECTOR_57  0x001C800 /* 2 Kb   */
+#define FLASH_SECTOR_58  0x001D000 /* 2 Kb   */
+#define FLASH_SECTOR_59  0x001D800 /* 2 Kb   */
+#define FLASH_SECTOR_60  0x001E000 /* 2 Kb   */
+#define FLASH_SECTOR_61  0x001E800 /* 2 Kb   */
+#define FLASH_SECTOR_62  0x001F000 /* 2 Kb   */
+#define FLASH_SECTOR_63  0x001F800 /* 2 Kb   */
+#define FLASH_TOP        0x0200000
```

Comment:
Defining the array is not needed if all the sectors have the same size. `flash_sector` below is not used.